CRM/Price/BAO/PriceField.php

```diff
@@ -598,7 +599,7 @@ public static function addQuickFormElement(
             }
             $opt['label'] = $preHelpText . $opt['label'] . $postHelpText;
           }
-          $priceVal = implode($seperator, array($opt[$valueFieldName] + $taxAmount, $count, $max_value));
+          $priceVal = implode($seperator, array($opt[$valueFieldName] + $taxAmount, $count, $max_value, $taxAmount));
```

Comment:
I'm a bit worried that we are making changes to generic pricefield BAO here just to support MembershipRenewal form.  What affect will this have elsewhere in Civi?

Reply:
@mattwire  I looked a bit more at this - there is a function in the BAO layer addQuickFormElement which is really a form helper. The function is on the toxic code list - where did the fishes go @totten ?

This priceVal winds up on the form as part of the element. 
```
        $element = &$qf->add('text', $elementName, $label,
          array_merge($extra,
            array(
              'price' => json_encode(array($optionKey, $priceVal)),
              'size' => '4',
            )
          ),
          $useRequired && $field->is_required
        );
```

This is probably a better model however

```
          $extra = array(
            'price' => json_encode(array($elementName, $priceVal)),
            'data-amount' => $opt[$valueFieldName],
            'data-currency' => $currencyName,
            'data-price-field-values' => json_encode($customOption),
            'visibility' => $visibility_id,
          );
....
          $choice[$opId] = $qf->createElement('radio', NULL, '', $opt['label'], $opt['id'], $extra);
````

Basically there was some work done at 'some point' adding data attributes to various price field entities, options & values allowing us to use the data attribute in js - eg like

```
/**
 * Calculate the total fee for the visible priceset.
 */
function calculateTotalFee() {
  var totalFee = 0;
  cj("#priceset [price]").each(function () {
    totalFee = totalFee + cj(this).data('line_raw_total');
  });
  return totalFee;
}
```
So this would be a way to get make it possible to do calculations at the js layer with appropriate tax information - I assume we should be assigning the tax rate not the tax amount & letting js layer do the work?